include/zephyr/net/net_if.h

```diff
@@ -332,6 +332,12 @@ struct net_if_dhcpv4 {
 
 	/** Number of attempts made for REQUEST and RENEWAL messages */
 	uint8_t attempts;
+
+	/** The address of the server the request is sent to */
+	struct in_addr request_server_addr;
```

Comment:
Do we need this address, would server_id field store the same data anyway?

Reply:
Yes I believe we do. The server_id field can get changed by a received response so then we wouldn't be comparing against the original request address.

Reply:
Ok, I was reading the code but it was not very clear whether it is overwritten or not.